test/tools/base64_test.cpp

```diff
@@ -18,69 +18,245 @@
 
-  add_test_with_context(suite, Base64, base64_encode_decode);
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodedStr);
+    ASSERT_EQ(decodedStr,inputStr6);
+};
+
+/**
+* Test Case: Unpadded input - 'missing' "=="
+*/
+TEST_F(Base64EncodeDecodeTests, base64_decode_unpaddedInput){
+    std::string origStr = "abcdefg";
+    
+
+    std::string encodeStr = uprotocol::tools::Base64::base64encode(origStr); //encoded str is YWJjZGVmZw==
+    std::string encodeStrNoPadding = "YWJjZGVmZw"; // Note the 'missing' "=="
+    ASSERT_NE(encodeStr, encodeStrNoPadding);
+
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodeStr);
+    ASSERT_EQ(decodedStr, origStr);
+
+    std::string decodedStrNoPadding = uprotocol::tools::Base64::base64decode(encodeStrNoPadding);
+    ASSERT_EQ(decodedStrNoPadding, origStr);
+};
+
+/**
+* Test Case: Tests single character input data
+*/
+TEST_F(Base64EncodeDecodeTests, input_SingleChar){
+    std::string origStr = "a";
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodedStr);
+    ASSERT_EQ(decodedStr, origStr);
+};
+
+/**
+* Test Case: Tests empty input data
+*/
+TEST_F(Base64EncodeDecodeTests, emptyInputData){
+    std::string origStr = "";
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodedStr);
+    ASSERT_EQ(decodedStr, origStr);
+};
+
+/**
+* Test Case: Tests Base64 encode and decode when passing invalid data length
+*/
+TEST_F(Base64EncodeDecodeTests, invalid_encoded_decoded_datalength){
+   
+    static const char origStr[] = "abcdefg";
+    size_t origStrLen = sizeof(origStr);
+    size_t origStrInvalidLen = 5;
+    ASSERT_NE(origStrLen, origStrInvalidLen);
+
+    std::string encodedStrVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                origStrLen);
+    std::string encodedStrInVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                origStrInvalidLen);
+    ASSERT_NE(encodedStrVal, encodedStrInVal);
+
+    size_t encodedStrInValLen = 5;
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                    encodedStrVal.length());
+    std::string decodedInValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                       encodedStrInValLen);
+                                                                    
+    ASSERT_NE(decodedValStr, decodedInValStr);
+    ASSERT_EQ(decodedValStr.data(), std::string(origStr));
+};
+
```

Comment:
Add test cases for the invalid length: 'invalid length > actual input array length'
- for base64encode(array, length)
- for base64decode(array, length)

Reply:
Added